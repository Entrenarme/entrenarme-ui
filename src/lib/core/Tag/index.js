// @flow
import * as React from 'react';
import styled from 'styled-components';
import Chip from '@material-ui/core/Chip';
import JssProvider from 'react-jss/lib/JssProvider';
import { create } from 'jss';
import { createGenerateClassName, jssPreset } from '@material-ui/core/styles';

import { regularFont } from '../../helpers/fonts';
import colors from '../../helpers/colors';

const generateClassName = createGenerateClassName({ productionPrefix: 'eui' });
const jss = create(jssPreset());

type ExtendedTagProps = {
  options: {
    bgcolor: string,
    color: string,
    xs: boolean,
  },
};

const ExtendedTag = styled(Chip)`
  height: 24px !important;
  font-family: ${regularFont} !important;
  font-size: ${(props: ExtendedTagProps) =>
    props.options && props.options.xs ? '0.5625rem' : '0.875rem'} !important;
  background-color: ${(props: ExtendedTagProps) =>
    props.options.bgcolor} !important;

  span {
    color: ${(props: ExtendedTagProps) => props.options.color};
  }

  @media only screen and (max-width: 767px) {
    font-size: 0.75rem !important;
  }
`;

type Props = {
  title: React.Node,
  bgcolor: string,
  color: string,
  xs: boolean,
};

const Tag = ({ title, bgcolor, color, xs }: Props) => {
  return (
    <JssProvider jss={jss} generateClassName={generateClassName}>
      <ExtendedTag label={title} options={{ bgcolor, color, xs }} />
    </JssProvider>
  );
};

Tag.defaultProps = {
  bgcolor: colors.gray.light,
  color: colors.text,
  xs: false,
};

export default Tag;
