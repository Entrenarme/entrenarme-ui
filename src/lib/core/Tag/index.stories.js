// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Tag from './index';

storiesOf('Tag', module).add(
  'default',
  withInfo('Tag')(() => <Tag title="Test" />),
);
