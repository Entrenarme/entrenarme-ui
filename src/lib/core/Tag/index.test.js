import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import Tag from './index';

test('renders "Tag" with title', () => {
  const { container } = render(<Tag title="Test" />);

  expect(container.firstChild).toMatchSnapshot();
});
