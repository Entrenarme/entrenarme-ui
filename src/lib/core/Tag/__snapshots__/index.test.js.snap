// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Tag" with title 1`] = `
<div
  role="button"
  class="MuiChip-root-1 sc-bdVaJa eHiRQO"
  tabindex="-1"
  options="[object Object]"
>
  <span
    class="MuiChip-label-14"
  >
    Test
  </span>
</div>
`;
